.github/workflows/test.yaml

```
@@ -1,15 +1,22 @@
+---
 name: Test
 
 on:
-  push:
   pull_request:
-    branches: [ main ]
 jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@master
-    - uses: actions/setup-go@v1
-      with:
-        go-version: '1.16'
-    - run: go test ./...
+      - uses: actions/checkout@main
+      - uses: actions/setup-go@v1
```

I think if we already touched this, we can update the actions/setup-go version
```suggestion
      - uses: actions/setup-go@v5
```